### ISSUE TITLE
sort-comp reporting a bad error message

### ISSUE THREAD

**Issue by mikecousins** [yannickcr/eslint-plugin-react#507]
![image](https://cloud.githubusercontent.com/assets/725614/13893894/f1ab929c-ed27-11e5-9957-4bc77375af9c.png)

As you can see getCourtesyOptions IS before loadProvinces.

**Comment by calmdev** [comment]
What worked for me and may work applied to the context here is making sure that all `get` methods are defined prior to any `load` methods.

**Comment by neiker** [comment]
Same issue, same fix

**Comment by yannickcr** [comment]
Can you post your eslint configuration and a full code example that trigger the error? Thanks.

**Comment by neiker** [comment]
Yes, no problem:
`
{
  "parser": "babel-eslint",
  "extends": [
    "airbnb"
  ],
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true,
      "experimentalObjectRestSpread": true,
      "classes": true
    }
  },
  "rules": {
    "react-native/no-unused-styles": 2,
    "react-native/split-platform-components": 2,
    "react-native/no-inline-styles": 2,
    "react-native/no-color-literals": 2,
    "react/jsx-no-bind": [2, {
      "ignoreRefs": false,
      "allowArrowFunctions": false,
      "allowBind": false
    }],
    "complexity": [
      2,
      10
    ],
    "max-depth": [
      2,
      4
    ],
    "max-nested-callbacks": [
      2,
      2
    ],
    "max-params": [
      2,
      4
    ],
    "max-statements": [
      2,
      18
    ]
  },
  "globals": {
    "__DEV__": true
  },
  "plugins": [
    "react-native"
  ]
}
`

**Comment by yannickcr** [comment]
Thanks. Can you also post a React Component that trigger the bug?

**Comment by neiker** [comment]
I could not find it, but I remember it was something like:

```
constructor()
shouldComponentUpdate()
setSomething()
onDragStart()
getBound()
onSomething()
```

**Comment by calmdev** [comment]
I haven't been able to intentionally reproduce it either. It happen while I was refactoring a component. If I come across it again I'll post an example.

**Comment by mikecousins** [comment]
Yes, I can no longer reproduce this. Maybe it was fixed by accident?

**Comment by calmdev** [comment]
This is by no means a practical example, but it will cause the message to appear at least. 

Maybe it will help.

```js
export default class View extends React.Component {
  getA() {}
  componentDidMountOk() {} // message doesn't show using `componentDidMount`
  getB() {}
  componentWillMount() {}
  render() {}
}
```

![screen shot 2016-07-27 at 5 18 44 pm](https://cloud.githubusercontent.com/assets/461249/17192828/35483a3e-541e-11e6-9d60-0e9ce5095395.png)

**Comment by yannickcr** [comment]
Thanks @calmdev , I was able to reproduce the bug with your example. I'll fix this.
